framework/src/org/checkerframework/framework/util/expressionannotations/ExpressionAnnotationError.java

```diff
@@ -0,0 +1,82 @@
+package org.checkerframework.framework.util.expressionannotations;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.checkerframework.framework.source.Result;
+import org.checkerframework.framework.util.FlowExpressionParseUtil.FlowExpressionParseException;
+
+/** Helper class for creating expression annotation error strings. */
+public class ExpressionAnnotationError {
+    private static final String formatString = "[error for expression: %s error: %s]";
+    private static final Pattern p =
+            Pattern.compile("\\[error for expression: (.*) error: (.*)\\]");
+
+    /**
+     * Returns whether or not the given expression string is an error. That is whether it is a
+     * string that was generate by this class.
+     *
+     * @param expression expression string to test
+     * @return wheter or not the given expressions string is an error
+     */
+    public static boolean isExpressionError(String expression) {
+        Matcher matcher = p.matcher(expression);
+        return matcher.matches() && matcher.groupCount() == 2;
+    }
+
+    public final String expression;
+    public final String error;
+
+    public ExpressionAnnotationError(String expression, String error) {
+        this.expression = expression;
+        this.error = error;
+    }
+
+    public ExpressionAnnotationError(String expression, FlowExpressionParseException e) {
+        StringBuffer buf = new StringBuffer();
+        List<Result.DiagMessage> msgs = e.getResult().getDiagMessages();
+
+        for (Result.DiagMessage msg : msgs) {
+            buf.append(msg.getArgs()[0]);
+        }
+        this.error = buf.toString();
+        this.expression = expression;
+    }
+
+    public ExpressionAnnotationError(String error) {
+        Matcher matcher = p.matcher(error);
```

Comment:
You could just use `isExpressionError` here and remove the duplication of logic.